File: 2383. Minimum Hours of Training to Win a Competition/Solution.py
from typing import List


class Solution:
    def minNumberOfHours(
        self, initialEnergy: int, initialExperience: int, energy: List[int], experience: List[int]
    ) -> int:
        res = max(sum(energy) + 1 - initialEnergy, 0)
        for exp in experience:
            if exp >= initialExperience:
                res += exp + 1 - initialExperience
                initialExperience = exp + 1
            initialExperience += exp
        return res

File: 2383. Minimum Hours of Training to Win a Competition/Solution.java
import java.util.Arrays;

class Solution {
    public int minNumberOfHours(int initialEnergy, int initialExperience, int[] energy, int[] experience) {
        int res = Math.max(Arrays.stream(energy).sum() + 1 - initialEnergy, 0);
        for (int exp : experience) {
            if (exp >= initialExperience) {
                res += exp + 1 - initialExperience;
                initialExperience = exp + 1;
            }
            initialExperience += exp;
        }
        return res;
    }
}
